src/qhttp/Response.cc

```diff
@@ -142,28 +147,46 @@ void Response::send(std::string const& content, std::string const& contentType)
     headers["Content-Type"] = contentType;
     headers["Content-Length"] = std::to_string(content.length());
 
-    std::ostream responseStream(&_responsebuf);
-    responseStream << _headers() << "\r\n" << content;
-    _write();
+    _responseBuf = _headers() + "\r\n" + content;
+    _startTransmit();
+    asio::async_write(*_socket, asio::buffer(_responseBuf.data(), _responseBuf.size()),
+                      [self = shared_from_this()](boost::system::error_code const& ec, std::size_t sent) {
+                          self->_finishTransmit(ec, sent);
+                      });
 }
 
 void Response::sendFile(fs::path const& path) {
+    _bytesToRead = fs::file_size(path);
```

Comment:
`_bytesRemaining` might be slightly better name?  Or maybe `_bytesToRead` as the full file size (never changes) and `_bytesRemaining` to track how much remains to be sent?

Reply:
Good point. Will fix it.